pkg/route/ingress/ingress.go

```diff
@@ -583,14 +583,16 @@ func targetPortForService(namespace string, path *extensionsv1beta1.HTTPIngressP
 		expect := path.Backend.ServicePort.StrVal
 		for _, port := range service.Spec.Ports {
 			if port.Name == expect {
-				return &port.TargetPort
+				targetPort := intstr.FromString(port.Name)
+				return &targetPort
 			}
 		}
 	} else {
 		for _, port := range service.Spec.Ports {
 			expect := path.Backend.ServicePort.IntVal
 			if port.Port == expect {
-				return &port.TargetPort
+				targetPort := intstr.FromInt(int(port.Port))
+				return &targetPort
```

Comment:
See https://github.com/openshift/origin/pull/23951/files#r337271800.

Reply:
I have addressed this issue in 5113ff5.

This changes the int port handling back to the way it was and only addresses the named port issues.

There are two cases for named Pod ports:
1. the Service has a named port which references a named port in the Pod. This is resolved by my commit.
2. the Service has an unnamed port which references a named port in the Pod. **This is still open** - I have added a test case for that `"create route - targetPort string, service port has no name"` but the implementation is a bit too much for me. In this case the port has to read from the corresponding Endpoint and set in the Route.